src/services/StorageService.ts

```diff
@@ -0,0 +1,97 @@
+import AWS from 'aws-sdk';
+
+const s3 = new AWS.S3({ region: process.env.BUCKET_REGION });
+export class StorageService {
+  async uploadFile(
+    fileName: string,
+    file: Express.Multer.File,
+    options?: Object
+  ): Promise<string | null> {
+    let fileNameKey = fileName;
+    if (options) {
+      if ('appendFileName' in options) {
+        fileNameKey = `${fileName}_${options['appendFileName']}`;
+      }
+    }
+    const params = {
+      Bucket: process.env.BUCKET_NAME,
+      Key: fileNameKey,
+      Body: file.buffer,
+      ContentDisposition: `attachment; filename="${fileName}${file.originalname.substring(
+        file.originalname.lastIndexOf('.')
+      )}"`,
+      ContentType: `${file.mimetype}`,
+    };
+
+    return s3
+      .upload(params)
+      .promise()

```

Comment:
is it possible to switch to await here?

Reply:
do we want to be handling errors in the services or on the controllers level?
we should await these calls where we want to use a try catch to output errors